### PR TITLE
docs: adds semantic HTML accessibility documentation

### PR DESCRIPTION
**Pull request by lindseywild**
Closes https://github.com/github/accessibility/issues/750.

Adds documentation about using semantic HTML, why it's important, how to test, etc.

Please check for correctness, grammar, and if you feel anything is missing. Thanks!

### CODE REVIEW

File: content/accessibility/semantic-html.mdx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+- Linting
+    - [Primer ViewComponents](https://github.com/primer/view_components/blob/main/docs/content/linting.md)
+    - [Primer React](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#linting)
+    - [eslint-plugin-jsx-a11y](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y)
+
+## Guidelines
+
+### For designers
+
+In your designs, annotate what HTML elements should be used for various parts of the design, if appropriate. Understand that most designs are achievable with CSS alone, agnostic of the HTML elements used. If a component has state, annotate the ARIA attributes needed for each state. For example, for an accordion, indicate that the header should have `aria-expanded="true"` when the panel is expanded, and `aria-expanded="false"` when collapsed.
```
Comment (inkblotty) [inline content/accessibility/semantic-html.mdx:32]: Can true semantics indicate state? I'm waffling here on whether to dive into `ARIA` examples on this Semantics overview.

**Review by inkblotty — APPROVED** [review]
Non-blocking thought -- let me know what you think.

File: content/accessibility/semantic-html.mdx (hunk 1 above)
Comment (lindseywild) [reply content/accessibility/semantic-html.mdx:32]: Thanks for bring this up! Now re-reading it with your feedback, I think I should take that part out. I feel it's not really semantic related, but more of a best practice when using ARIA.

Comment (inkblotty) [reply content/accessibility/semantic-html.mdx:32]: Ok 👍

**Comment by lindseywild** [comment]
@primer/reviewers could someone take a look at this, please and thanks!

**Review by langermank — APPROVED** [review]

**Pull request merged**